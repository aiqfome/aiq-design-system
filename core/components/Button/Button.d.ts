import React, { ReactNode } from 'react';
import { DefaultTheme } from 'styled-components';
import { SpaceProps, BorderProps, LayoutProps, FontSizeProps, FontWeightProps } from 'styled-system';
export interface Props extends DefaultTheme, SpaceProps, LayoutProps, BorderProps, FontSizeProps, FontWeightProps {
    children?: ReactNode;
    prefix?: any;
    sufix?: any;
    refButton?: any;
    variantType?: string;
    variant?: 'text' | 'contained' | 'outlined' | 'fab' | 'icon';
    palette?: 'primary' | 'error' | 'secondary' | 'neutral';
    onClick?: any;
    fullWidth?: boolean;
    disabled?: boolean;
    className?: string;
    type?: 'button' | 'submit' | 'reset' | undefined;
}
export declare const ButtonStyled: import("styled-components").StyledComponent<"button", any, Props, never>;
export declare const Button: React.FC<Props>;
//# sourceMappingURL=Button.d.ts.map