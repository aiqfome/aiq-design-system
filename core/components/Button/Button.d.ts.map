{"version":3,"file":"Button.d.ts","sourceRoot":"","sources":["../../../src/components/Button/Button.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,SAAS,EAAe,MAAM,OAAO,CAAA;AAErD,OAAe,EAAO,YAAY,EAAE,MAAM,mBAAmB,CAAA;AAC7D,OAAO,EAGL,UAAU,EAEV,WAAW,EAEX,WAAW,EAEX,aAAa,EAEb,eAAe,EAChB,MAAM,eAAe,CAAA;AAKtB,MAAM,WAAW,KACf,SAAQ,YAAY,EAClB,UAAU,EACV,WAAW,EACX,WAAW,EACX,aAAa,EACb,eAAe;IACjB,QAAQ,CAAC,EAAE,SAAS,CAAA;IACpB,MAAM,CAAC,EAAE,GAAG,CAAA;IACZ,KAAK,CAAC,EAAE,GAAG,CAAA;IACX,SAAS,CAAC,EAAE,GAAG,CAAA;IACf,WAAW,CAAC,EAAE,MAAM,CAAA;IACpB,OAAO,CAAC,EAAE,MAAM,GAAG,WAAW,GAAG,UAAU,GAAG,KAAK,GAAG,MAAM,CAAA;IAC5D,OAAO,CAAC,EAAE,SAAS,GAAG,OAAO,GAAG,WAAW,GAAG,SAAS,CAAA;IACvD,OAAO,CAAC,EAAE,GAAG,CAAA;IACb,SAAS,CAAC,EAAE,OAAO,CAAA;IACnB,QAAQ,CAAC,EAAE,OAAO,CAAA;IAClB,SAAS,CAAC,EAAE,MAAM,CAAA;IAClB,IAAI,CAAC,EAAE,QAAQ,GAAG,QAAQ,GAAG,OAAO,GAAG,SAAS,CAAA;CACjD;AAED,eAAO,MAAM,YAAY,0EAsNxB,CAAA;AAED,eAAO,MAAM,MAAM,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CA0DlC,CAAA"}