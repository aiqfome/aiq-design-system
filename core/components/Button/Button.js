"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Button = exports.ButtonStyled = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
const Icon_1 = require("../Icon");
const Text_1 = require("../Text");
exports.ButtonStyled = styled_components_1.default.button `
  border-radius: 5px;
  display: flex;
  align-items: center;
  padding: 10px 20px;
  max-height: 42px;
  cursor: pointer;

  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.border}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}

  &:active {
    opacity: 0.5;
  }

  ${({ disabled }) => disabled &&
    styled_components_1.css `
      opacity: 0.5;

      &:hover {
        cursor: not-allowed;
      }
    `}

  ${({ fullWidth }) => fullWidth &&
    styled_components_1.css `
      width: 100%;
    `}

    
  &.__button-text {
    border: none;
    background: none;
    padding: 0;
    text-align: left;

    &:hover {
      text-decoration: underline;
    }

    &.__button-text-primary {
      color: ${({ theme }) => theme.colors.primary};
    }

    &.__button-text-error {
      color: ${({ theme }) => theme.colors.error};
    }

    &.__button-text-secondary {
      color: ${({ theme }) => theme.colors.primary};
    }

    &.__button-text-neutral {
      color: ${({ theme }) => theme.colors.almostBlack};
    }
  }

  &.__button-contained {
    text-transform: uppercase;
    border: none;
    justify-content: center;
    transition: 0.5s;

    &.__button-contained-primary {
      color: ${({ theme }) => theme.colors.white};
      background: ${({ theme }) => theme.colors.primary};

      &:hover {
        background: ${({ theme }) => theme.colors.primaryMedium};
      }
    }

    &.__button-contained-error {
      color: ${({ theme }) => theme.colors.white};
      background: ${({ theme }) => theme.colors.error};

      &:hover {
        background: ${({ theme }) => theme.colors.error};
      }
    }

    &.__button-contained-secondary {
      color: ${({ theme }) => theme.colors.white};
      background: ${({ theme }) => theme.colors.secondary};

      &:hover {
        background: ${({ theme }) => theme.colors.secondaryMedium};
      }
    }

    &.__button-contained-neutral {
      background: ${({ theme }) => theme.colors.white};
      color: ${({ theme }) => theme.colors.almosBlack};
      border: 1px solid ${({ theme }) => theme.colors.almostBlack};
    }
  }

  &.__button-outlined {
    text-transform: uppercase;
    justify-content: center;
    transition: 0.5s;

    &.__button-outlined-primary {
      border: 1px solid ${({ theme }) => theme.colors.primary};
      color: ${({ theme }) => theme.colors.primary};
      background: none;

      &:hover {
        background: ${({ theme }) => theme.colors.primary};
        color: ${({ theme }) => theme.colors.white};
      }
    }

    &.__button-outlined-error {
      border: 1px solid ${({ theme }) => theme.colors.error};
      color: ${({ theme }) => theme.colors.error};
      background: none;

      &:hover {
        background: ${({ theme }) => theme.colors.error};
        color: ${({ theme }) => theme.colors.white};
      }
    }

    &.__button-outlined-secondary {
      border: 1px solid ${({ theme }) => theme.colors.secondary};
      color: ${({ theme }) => theme.colors.secondary};
      background: none;

      &:hover {
        background-color: ${({ theme }) => theme.colors.secondaryDark};
      }
    }

    &.__button-outlined-neutral {
      border: 1px solid ${({ theme }) => theme.colors.mediumGrey};
      color: ${({ theme }) => theme.colors.almostBlack};
      background: ${({ theme }) => theme.colors.white};

      &:hover {
        background-color: ${({ theme }) => theme.colors.lightGrey};
      }
    }
  }

  &.__button-fab {
    text-transform: uppercase;
    justify-content: center;
    border-radius: 24px;
    border: none;
    color: ${({ theme }) => theme.colors.white};
    padding: 14px 21px;
    transition: 0.5s;

    &.__button-fab-icon {
      border-radius: 50%;
      padding: 0;
      min-height: 50px;
      min-width: 50px;
    }

    &.__button-fab-primary {
      background: ${({ theme }) => theme.colors.primary};
    }

    &.__button-fab-error {
      background: ${({ theme }) => theme.colors.error};
    }

    &.__button-fab-secondary {
      background: ${({ theme }) => theme.colors.secondary};
    }

    &.__button-fab-neutral {
      color: ${({ theme }) => theme.colors.almostBlack};
      background: ${({ theme }) => theme.colors.white};
    }
  }

  &.__button-icon {
    text-transform: uppercase;
    justify-content: center;
    border: none;
    background: none;
    padding: 8px;
    border-radius: 50%;
    transition: 0.5s;

    background-position: center;
    transition: background 0.5s;

    &:hover {
      background: ${({ theme }) => theme.colors.lightGrey}
        radial-gradient(
          circle,
          transparent 1%,
          ${({ theme }) => theme.colors.lightGrey} 1%
        )
        center/15000%;
    }

    &:active {
      background-color: ${({ theme }) => theme.colors.grey};
      background-size: 100%;
      transition: background 0s;
      opacity: 0.5;
    }
  }
`;
exports.Button = (_a) => {
    var { children, prefix, sufix, palette = 'primary', variant = 'text', variantType, type = 'button', className } = _a, props = __rest(_a, ["children", "prefix", "sufix", "palette", "variant", "variantType", "type", "className"]);
    const getClassName = react_1.useCallback(() => {
        return `${className} __button-${variant} __button-${variant}-${palette} ${variantType && `__button-${variant}-${variantType}`}`;
    }, [palette, variant, variantType, className]);
    if (sufix && prefix) {
        return (react_1.default.createElement(exports.ButtonStyled, Object.assign({ className: getClassName(), type: type }, props),
            react_1.default.createElement(Icon_1.Icon, { "data-testid": 'button-prefix', cursor: 'pointer', mr: 5 }, prefix),
            react_1.default.createElement(Text_1.Text, { fontSize: 'medium' }, children),
            react_1.default.createElement(Icon_1.Icon, { "data-testid": 'button-sufix', cursor: 'pointer', ml: 5 }, sufix)));
    }
    if (prefix) {
        return (react_1.default.createElement(exports.ButtonStyled, Object.assign({ className: getClassName(), type: type }, props),
            react_1.default.createElement(Icon_1.Icon, { "data-testid": 'button-prefix', cursor: 'pointer', mr: 5 }, prefix),
            react_1.default.createElement(Text_1.Text, { fontSize: 'medium' }, children)));
    }
    if (sufix) {
        return (react_1.default.createElement(exports.ButtonStyled, Object.assign({ className: getClassName(), type: type }, props),
            react_1.default.createElement(Text_1.Text, { fontSize: 'medium' }, children),
            react_1.default.createElement(Icon_1.Icon, { "data-testid": 'button-sufix', cursor: 'pointer', ml: 5 }, sufix)));
    }
    return (react_1.default.createElement(exports.ButtonStyled, Object.assign({ type: type, className: getClassName() }, props),
        react_1.default.createElement(Text_1.Text, { fontSize: 'medium' }, children)));
};
exports.Button.propTypes = {
    children: prop_types_1.default.any,
    prefix: prop_types_1.default.any,
    className: prop_types_1.default.string,
    sufix: prop_types_1.default.any,
    refButton: prop_types_1.default.any,
    palette: prop_types_1.default.oneOf(['primary', 'error', 'secondary', 'neutral']),
    variantType: prop_types_1.default.string,
    variant: prop_types_1.default.oneOf(['text', 'contained', 'outlined', 'fab', 'icon']),
    disabled: prop_types_1.default.bool,
    fontWeight: prop_types_1.default.oneOfType([prop_types_1.default.string, prop_types_1.default.number]),
    onClick: prop_types_1.default.func,
    type: prop_types_1.default.oneOf(['button', 'submit', 'reset'])
};
