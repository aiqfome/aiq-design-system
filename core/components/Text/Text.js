"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
exports.Text = styled_components_1.default.span `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.typography}

  white-space: ${props => props.whiteSpace || 'normal'};

  ${({ cursor }) => cursor &&
    styled_components_1.css `
      cursor: ${cursor};
    `};

  ${({ truncate }) => truncate &&
    styled_components_1.css `
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `};
`;
exports.Text.propTypes = {
    cursor: prop_types_1.default.string,
    whiteSpace: prop_types_1.default.oneOf(['nowrap', 'normal', 'pre'])
};
