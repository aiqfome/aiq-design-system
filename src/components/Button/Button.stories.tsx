import React, { ReactElement } from 'react'
import { withKnobs, text, select, boolean } from '@storybook/addon-knobs'

import { FaHamburger } from 'react-icons/fa'

import { Button } from './Button'
import { Flex } from '../Flex'

export default {
  component: Button,
  title: 'Button',
  decorators: [withKnobs as any]
}

export const Text: React.FC = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Button
      variant='text'
      palette={select(
        'Palette',
        {
          Primary: 'primary',
          Secondary: 'secondary',
          Neutral: 'neutral'
        },
        'primary'
      )}
      fontWeight={select(
        'Font Weight',
        {
          Regular: 'regular',
          Medium: 'medium',
          SemiBold: 'semiBold',
          Bold: 'bold'
        },
        'regular'
      )}
    >
      {text('Label', 'aiqfome!')}
    </Button>
  </Flex>
)

export const Contained: React.FC = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Flex
      variant='centralized'
      height='200px'
      width='400px'
      border='1px solid lightGrey'
    >
      <Button
        fullWidth={boolean('Full Width', false)}
        variant='contained'
        palette={select(
          'Palette',
          {
            Primary: 'primary',
            Secondary: 'secondary',
            Neutral: 'neutral'
          },
          'primary'
        )}
        fontWeight={select(
          'Font Weight',
          {
            Regular: 'regular',
            Medium: 'medium',
            SemiBold: 'semiBold',
            Bold: 'bold'
          },
          'regular'
        )}
      >
        {text('Label', 'aiqfome!')}
      </Button>
    </Flex>
  </Flex>
)

export const Disabled: React.FC = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Flex
      variant='centralized'
      height='200px'
      width='400px'
      border='1px solid lightGrey'
    >
      <Button
        fullWidth={boolean('Full Width', false)}
        variant='contained'
        mx={6}
        disabled={true}
        palette={'primary'}
      >
        contained
      </Button>
      <Button
        fullWidth={boolean('Full Width', false)}
        variant='outlined'
        mx={6}
        disabled={true}
        palette={'primary'}
      >
        outlined
      </Button>
      <Button
        fullWidth={boolean('Full Width', false)}
        variant='fab'
        mx={6}
        disabled={true}
        palette={'primary'}
      >
        fab
      </Button>
    </Flex>
  </Flex>
)

export const Outlined: React.FC = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Flex
      variant='centralized'
      height='200px'
      width='400px'
      border='1px solid lightGrey'
    >
      <Button
        variant='outlined'
        fullWidth={boolean('Full Width', false)}
        palette={select(
          'Palette',
          {
            Primary: 'primary',
            Secondary: 'secondary',
            Neutral: 'neutral'
          },
          'primary'
        )}
        fontWeight={select(
          'Font Weight',
          {
            Regular: 'regular',
            Medium: 'medium',
            SemiBold: 'semiBold',
            Bold: 'bold'
          },
          'regular'
        )}
      >
        {text('Label', 'aiqfome!')}
      </Button>
    </Flex>
  </Flex>
)

export const Fab: React.FC = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Flex
      variant='centralized'
      height='200px'
      width='400px'
      border='1px solid lightGrey'
    >
      <Button
        variant='fab'
        fullWidth={boolean('Full Width', false)}
        palette={select(
          'Palette',
          {
            Primary: 'primary',
            Secondary: 'secondary',
            Neutral: 'neutral'
          },
          'primary'
        )}
        fontWeight={select(
          'Font Weight',
          {
            Regular: 'regular',
            Medium: 'medium',
            SemiBold: 'semiBold',
            Bold: 'bold'
          },
          'regular'
        )}
      >
        {text('Label', 'aiqfome!')}
      </Button>
    </Flex>
  </Flex>
)

export const FabIcon: React.FC = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Button
      variant='fab'
      variantType='icon'
      palette={select(
        'Palette',
        {
          Primary: 'primary',
          Secondary: 'secondary',
          Neutral: 'neutral'
        },
        'primary'
      )}
    >
      <FaHamburger />
    </Button>
  </Flex>
)

export const Prefix: React.FC = (): ReactElement => (
  <Flex variant='fullCentralized'>
    <Flex
      variant='centralized'
      height='200px'
      width='400px'
      border='1px solid lightGrey'
    >
      <Button
        fullWidth={boolean('Full Width', false)}
        palette={select(
          'Palette',
          {
            Primary: 'primary',
            Secondary: 'secondary',
            Neutral: 'neutral'
          },
          'primary'
        )}
        variant={select(
          'variant',
          {
            Text: 'text',
            Contained: 'contained',
            Outlined: 'outlined',
            Fab: 'fab'
          },
          'text'
        )}
        prefix={<FaHamburger />}
      >
        {text('Label', 'aiqfome!')}
      </Button>
    </Flex>
  </Flex>
)

export const ClickAction: React.FC = (): ReactElement => {
  const handleClick = () => {
    alert('i really want a pizza')
  }

  return (
    <Flex variant='fullCentralized'>
      <Flex
        variant='centralized'
        height='200px'
        width='400px'
        border='1px solid lightGrey'
      >
        <Button
          onClick={() => handleClick()}
          fullWidth={boolean('Full Width', false)}
          palette={select(
            'Palette',
            {
              Primary: 'primary',
              Secondary: 'secondary',
              Neutral: 'neutral'
            },
            'primary'
          )}
          variant={select(
            'variant',
            {
              Text: 'text',
              Contained: 'contained',
              Outlined: 'outlined',
              Fab: 'fab'
            },
            'contained'
          )}
          sufix={<FaHamburger />}
        >
          {text('Label', 'click me!')}
        </Button>
      </Flex>
    </Flex>
  )
}
