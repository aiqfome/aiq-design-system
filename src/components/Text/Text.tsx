import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  typography,
  TypographyProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    TypographyProps {
  cursor?: string
  whiteSpace?: 'nowrap' | 'normal' | 'pre'
}

export const Text = styled.span<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${typography}

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: ${cursor};
    `};
  white-space: ${props => props.whiteSpace || 'normal'};
`

Text.propTypes = {
  cursor: PropTypes.string,
  whiteSpace: PropTypes.oneOf(['nowrap', 'normal', 'pre'])
}
