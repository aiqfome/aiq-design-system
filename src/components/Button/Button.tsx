import React, { ReactNode } from 'react'
import PropTypes from 'prop-types'
import styled, { css, DefaultTheme } from 'styled-components'
import {
  color,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps
} from 'styled-system'

import { Icon } from '../Icon'
import { Text } from '../Text'

export interface Props
  extends DefaultTheme,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps {
  children?: ReactNode
  prefix?: any
  sufix?: any
  refButton?: any
  variantType?: string
  variant?: 'text' | 'contained' | 'outlined' | 'fab'
  palette?: 'primary' | 'secondary' | 'neutral'
  onClick?: any
  fullWidth?: boolean
  type?: 'button' | 'submit' | 'reset' | undefined
}

const buttonVariations: { [index: string]: any } = {
  text: css<Props>`
    border: none;
    background: none;
    padding: 0;

    &:hover {
      text-decoration: underline;
    }

    ${({ palette }) =>
      palette === 'primary' &&
      css`
        color: ${({ theme }) => theme.colors.primary};
      `}

    ${({ palette }) =>
      palette === 'secondary' &&
      css`
        color: ${({ theme }) => theme.colors.secondary};
      `}
    ${({ palette }) =>
      palette === 'neutral' &&
      css`
        color: ${({ theme }) => theme.colors.almosBlack};
      `}
  `,
  contained: css<Props>`
    border: none;

    ${({ palette }) =>
      palette === 'primary' &&
      css`
        color: ${({ theme }) => theme.colors.white};
        background: ${({ theme }) => theme.colors.primary};

        &:hover {
          background: ${({ theme }) => theme.colors.primaryMedium};
        }
      `};
    ${({ palette }) =>
      palette === 'secondary' &&
      css`
        color: ${({ theme }) => theme.colors.white};
        background: ${({ theme }) => theme.colors.secondary};

        &:hover {
          background: ${({ theme }) => theme.colors.secondaryMedium};
        }
      `};
    ${({ palette }) =>
      palette === 'neutral' &&
      css`
        background: ${({ theme }) => theme.colors.white};
        color: ${({ theme }) => theme.colors.almosBlack};
        border: 1px solid ${({ theme }) => theme.colors.almostBlack};
      `}
  `,
  outlined: css<Props>`
    ${({ palette }) =>
      palette === 'primary' &&
      css`
        border: 1px solid ${({ theme }) => theme.colors.primary};
        color: ${({ theme }) => theme.colors.primary};
        background: none;

        &:hover {
          background: ${({ theme }) => theme.colors.primaryLighest};
        }
      `}
    ${({ palette }) =>
      palette === 'secondary' &&
      css`
        border: 1px solid ${({ theme }) => theme.colors.secondary};
        color: ${({ theme }) => theme.colors.secondary};
        background: none;

        &:hover {
          background-color: ${({ theme }) => theme.colors.secondaryLighest};
        }
      `}
    ${({ palette }) =>
      palette === 'neutral' &&
      css`
        border: 1px solid ${({ theme }) => theme.colors.mediumGrey};
        color: ${({ theme }) => theme.colors.almostBlack};
        background: ${({ theme }) => theme.colors.white};

        &:hover {
          background-color: ${({ theme }) => theme.colors.lightGrey};
        }
      `}
  `,
  fab: css<Props>`
    border-radius: 24px;
    border: none;
    color: ${({ theme }) => theme.colors.white};
    padding: 14px 21px;

    ${({ variantType }) =>
      variantType === 'icon' &&
      css`
        border-radius: 50%;
        padding: 0;
        min-height: 50px;
        min-width: 50px;
      `}
    ${({ palette }) =>
      palette === 'primary' &&
      css`
        background: ${({ theme }) => theme.colors.primary};
      `}
    ${({ palette }) =>
      palette === 'secondary' &&
      css`
        background: ${({ theme }) => theme.colors.secondary};
      `}
    ${({ palette }) =>
      palette === 'neutral' &&
      css`
        color: ${({ theme }) => theme.colors.almostBlack};
        background: ${({ theme }) => theme.colors.white};
      `}
  `
}

export const ButtonStyled = styled.button<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}

  border-radius: 5px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 10px 20px;
  max-height: 42px;
  cursor: pointer;

  ${({ fullWidth }) =>
    fullWidth &&
    css`
      width: 100%;
    `}

  ${({ variant }) => buttonVariations[variant || 'text']}
`

export const Button: React.FC<Props> = ({
  children,
  prefix,
  sufix,
  type = 'button',
  ...props
}) => {
  if (prefix) {
    return (
      <ButtonStyled type={type} {...props}>
        <Icon cursor='pointer' mr={5}>
          {prefix}
        </Icon>
        <Text cursor='pointer' fontSize='medium'>
          {children}
        </Text>
      </ButtonStyled>
    )
  }

  if (sufix) {
    return (
      <ButtonStyled type={type} {...props}>
        <Text cursor='pointer' fontSize='medium'>
          {children}
        </Text>
        <Icon cursor='pointer' ml={5}>
          {sufix}
        </Icon>
      </ButtonStyled>
    )
  }

  return (
    <ButtonStyled type={type} {...props}>
      <Text cursor='pointer' fontSize='medium'>
        {children}
      </Text>
    </ButtonStyled>
  )
}

Button.propTypes = {
  children: PropTypes.any,
  prefix: PropTypes.any,
  sufix: PropTypes.any,
  refButton: PropTypes.any,
  fontWeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['button', 'submit', 'reset'])
}
