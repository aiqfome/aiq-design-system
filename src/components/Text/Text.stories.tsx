import React from 'react'
import { withKnobs, text, number } from '@storybook/addon-knobs'

import { Text } from './Text'
import { Flex } from '../Flex'

export default {
  component: Text,
  title: 'Text',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Text
    color={text('color', 'primary')}
    fontSize={number('fontSize', 16)}
    cursor={text('cursor', '')}
    fontWeight={text('fontWeight', 'medium')}
  >
    {text('text', 'Aiqfome')}
  </Text>
)

export const cursor: React.FC = () => (
  <Text cursor='pointer'>Design System</Text>
)

export const WhiteSpace: React.FC = () => (
  <Flex width='50px' border='1px solid black'>
    <Text whiteSpace='nowrap'>Design System</Text>
  </Flex>
)
