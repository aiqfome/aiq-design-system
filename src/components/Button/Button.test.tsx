import React from 'react'

import { Button } from './Button'
import { Text } from '../Text'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(
      <Button>
        <Text>Button component!</Text>
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
