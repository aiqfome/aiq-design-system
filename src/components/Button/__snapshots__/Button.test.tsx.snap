// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<button
  className="sc-AxhCb jGTOcA"
>
  <span
    className="sc-AxiKw gjPZme"
    cursor="pointer"
    fontSize="medium"
  >
    <span
      className="sc-AxiKw bBViMC"
    >
      Button component!
    </span>
  </span>
</button>
`;
